static/app/components/modals/widgetBuilder/addToDashboardModal.tsx

```diff
@@ -43,13 +72,40 @@ function AddToDashboardModal({
   }, []);
 
   function handleGoToBuilder() {
+    const pathname =
+      selectedDashboardId === 'new'
+        ? `/organizations/${organization.slug}/dashboards/new/widget/new/`
+        : `/organizations/${organization.slug}/dashboard/${selectedDashboardId}/widget/new/`;
+
+    router.push({
+      pathname,
+      query: widgetAsQueryParams,
+    });
     closeModal();
-    return;
   }
 
-  function handleAddAndStayInDiscover() {
-    closeModal();
-    return;
+  async function handleAddAndStayInDiscover() {
+    if (selectedDashboardId === null) {
+      return;
+    }
+
+    try {
+      const dashboard = await fetchDashboard(api, organization.slug, selectedDashboardId);
+      const newDashboard = {
+        ...dashboard,
+        widgets: [
+          ...dashboard.widgets,
+          {...widget, title: widget.title === '' ? 'All Events' : widget.title},
+        ],
+      };
+
+      await updateDashboard(api, organization.slug, newDashboard);
+
+      closeModal();
+      addSuccessMessage(t('Successfully added widget to dashboard'));
+    } catch (e) {
+      addErrorMessage(t('Unable to add widget to dashboard'));
+    }
```

Comment:
can we use the https://github.com/getsentry/sentry/blob/master/static/app/utils/handleXhrErrorResponse.tsx here?